src/AWS.Deploy.CLI/ServerMode/Controllers/DeploymentController.cs

```diff
@@ -127,6 +127,62 @@ public async Task<IActionResult> GetRecommendations(string sessionId)
             return Ok(output);
         }
 
+        /// <summary>
+        /// Gets the list of updatable option setting items for the selected recommendation.
+        /// </summary>
+        [HttpGet("session/<sessionId>/settings")]
+        [SwaggerOperation(OperationId = "GetConfigSettings")]
+        [SwaggerResponse(200, type: typeof(GetOptionSettingsOutput))]
+        [Authorize]
+        public IActionResult GetConfigSettings(string sessionId)
+        {
+            var state = _stateServer.Get(sessionId);
+            if (state == null)
+            {
+                return NotFound($"Session ID {sessionId} not found.");
+            }
+
+            if (state.SelectedRecommendation == null)
+            {
+                return NotFound($"A deployment target is not set for Session ID {sessionId}.");
+            }
+
+            var orchestrator = CreateOrchestrator(state);
+
+            var deploymentBundleDefinition = orchestrator.GetDeploymentBundleDefinition(state.SelectedRecommendation);
+
+            var configurableOptionSettings = state.SelectedRecommendation.Recipe.OptionSettings.Union(deploymentBundleDefinition.Parameters);
+
+            var output = new GetOptionSettingsOutput();
+
+            foreach (var setting in configurableOptionSettings)
+            {
+                if ((state.SelectedRecommendation.IsExistingCloudApplication && !setting.Updatable) || !state.SelectedRecommendation.IsOptionSettingDisplayable(setting))
+                    continue;
+
+                var settingSummary = new OptionSettingItemSummary(setting.Id, setting.Name, setting.Description, setting.Type.ToString())
+                {
+                    TypeHint = setting.TypeHint?.ToString(),
+                    Value = state.SelectedRecommendation.GetOptionSettingValue(setting),
+                    Advanced = setting.AdvancedSetting
+                };
+
+                setting.ChildOptionSettings.ForEach(childSetting =>
```

Comment:
Does this work if the option settings are more then 2 level deep? Meaning a object setting containing an object setting.

Reply:
We do not support that in general. An Object type can only have children of primitive types.

Reply:
This is not correct, our logic are written in way that it can handle any arbitrary depth. By chance, we only have 2 level settings in recipe.

Reply:
Added recursive logic to handle a depth more than 2